vita3k/modules/SceHttp/SceHttp.cpp

```diff
@@ -145,57 +502,456 @@ EXPORT(int, sceHttpGetIcmOption) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpGetLastErrno) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpGetLastErrno, int reqId, int *errNum) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (host.http.requests.find(reqId) == host.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    *errNum = (int)errno;
+
+    return 0;
 }
 
-EXPORT(int, sceHttpGetMemoryPoolStats) {
+EXPORT(int, sceHttpGetMemoryPoolStats, SceHttpMemoryPoolStats *currentStat) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!currentStat)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
     return UNIMPLEMENTED();
 }
 
 EXPORT(int, sceHttpGetNonblock) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpGetResponseContentLength) {
-    return UNIMPLEMENTED();
-}
+EXPORT(int, sceHttpGetResponseContentLength, int reqId, size_t *contentLength) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
 
-EXPORT(int, sceHttpGetStatusCode) {
-    return UNIMPLEMENTED();
-}
+    if (host.http.requests.find(reqId) == host.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = host.http.requests.find(reqId);
+
+    char contLen[15] = "Content-Length";
+
+    auto length_it = req->second.res.headers.find(contLen);
+    if (length_it == req->second.res.headers.end())
+        return RET_ERROR(SCE_HTTP_ERROR_NO_CONTENT_LENGTH);
+
+    auto len = length_it->second;
+    int length = std::stoi(len);
+    *contentLength = length;
 
-EXPORT(int, sceHttpInit) {
-    STUBBED("Fail when called");
-    return -1;
+    req->second.res.contentLength = length;
+
+    return 0;
 }
 
-EXPORT(int, sceHttpParseResponseHeader) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpGetStatusCode, int reqId, int *statusCode) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (host.http.requests.find(reqId) == host.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = host.http.requests.find(reqId)->second;
+
+    *statusCode = req.res.statusCode;
+    return 0;
 }
 
-EXPORT(int, sceHttpParseStatusLine) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpInit, size_t poolSize) {
+    if (host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_ALREADY_INITED);
+
+    if (poolSize == 0)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    STUBBED("ignore poolSize");
+
+    // TODO: what should SSL pool size be (2MB)?
+    CALL_EXPORT(sceSslInit, 2 * 1024 * 1024);
```

Comment:
Why is sceSslInit called in sceHttpInit? Games call sceHttpInit then sceSslInit. Doing this will make the second call return SCE_SSL_ERROR_ALREADY_INITED. 
By the way if you want a size indicator, Little Kings Story calls sceSslInit with pool size 0x4b000.

Reply:
Didn't really notice that lol